File: PHP/CycleORM/src/Models/Department.php
<?php

namespace App\Models;

use Cycle\Annotated\Annotation\Entity;
use Cycle\Annotated\Annotation\Column;

#[Entity(table: 'departments')]
class Department
{
    #[Column(type: 'primary')]
    public int $id;

    public function __construct(
        #[Column(type: 'string')]
        public string $name,
    ){}

    public function __toString(): string
    {
        return "{ Department { Id: $this->id, Name: $this->name } }";
    }
}

File: Scala/Slick/src/main/scala/models/Department.scala
package models

import slick.jdbc.SQLiteProfile.api.*

final case class Department(id: Option[Int], name: String):
  def this(name: String) =
    this(None, name)

  override def toString = s"Department { Id: ${id.getOrElse(0)}, Name: $name }"

class Departments(tag: Tag) extends Table[Department](tag, "department"):
  def id = column[Int]("id", O.PrimaryKey, O.AutoInc)
  def name = column[String]("name")
  def * = (id.?, name).mapTo[Department]

val departments = TableQuery[Departments]
